apps/antalmanac/src/components/RightPane/CoursePane/CoursePaneButtonRow.tsx

```diff
@@ -59,45 +59,33 @@ function renderEmptySelectValue() {
  * e.g. show/hide the section code, instructors, etc.
  */
 export function ColumnToggleButton() {
-    const [activeColumns, setActiveColumns] = useState(RightPaneStore.getActiveColumns());
+    const [ selectedColumns, setSelectedColumns ] = useColumnStore(store => [store.selectedColumns, store.setSelectedColumns]);
     const [open, setOpen] = useState(false);
 
-    const handleColumnChange = useCallback(
-        (newActiveColumns: SectionTableColumn[]) => {
-            logAnalytics({
-                category: analyticsEnum.classSearch.title,
-                action: analyticsEnum.classSearch.actions.TOGGLE_COLUMNS,
-            });
-
-            setActiveColumns(newActiveColumns);
-        },
-        [setActiveColumns]
-    );
-
-    const handleChange = useCallback(
-        (e: SelectChangeEvent<SectionTableColumn[]>) => {
-            if (typeof e.target.value !== 'string') {
-                RightPaneStore.setActiveColumns(e.target.value);
-            }
-        },
-        [RightPaneStore.setActiveColumns]
-    );
+    const handleChange = (e: SelectChangeEvent<SectionTableColumn[]>) => {
+        if (typeof e.target.value !== 'string') {
+            setSelectedColumns(e.target.value);
+        }
+    };
```

Comment:
For consistency, you can memoize this like the other handler functions.

```suggestion
    const handleChange = useCallback((e: SelectChangeEvent<SectionTableColumn[]>) => {
        if (typeof e.target.value !== 'string') {
            setSelectedColumns(e.target.value);
        }
    }, [])
```